File: Behavioural-Patterns/State/C#/State/Interfaces/OrderState.cs
public interface OrderState {
    
    void pay(Order order){
        Console.WriteLine("ERROR: Payment is not a valid action in the " + this.GetType().Name + " state.");
    }

    void confirmPayment(Order order) {
        Console.WriteLine("ERROR: Confirming payment is not a valid action in the " + this.GetType().Name + " state.");
    }

    void ship(Order order) {
        Console.WriteLine("ERROR: Shipping is not a valid action in the " + this.GetType().Name + " state.");
    }

    void deliver(Order order) {
        Console.WriteLine("ERROR: Delivering is not a valid action in the " + this.GetType().Name + " state.");
    }

    void cancel(Order order) {
        Console.WriteLine("ERROR: Cancelling is not a valid action in the " + this.GetType().Name + " state.");
    }
}

File: Behavioural-Patterns/State/Java/Interfaces/OrderState.java
package Interfaces;

import Implementations.Order;

public interface OrderState {
    
    default void pay(Order order){
        System.out.println("ERROR: Payment is not a valid action in the " + this.getClass().getSimpleName() + " state.");
    }

    default void confirmPayment(Order order) {
        System.out.println("ERROR: Confirming payment is not a valid action in the " + this.getClass().getSimpleName() + " state.");
    }

    default void ship(Order order) {
        System.out.println("ERROR: Shipping is not a valid action in the " + this.getClass().getSimpleName() + " state.");
    }

    default void deliver(Order order) {
        System.out.println("ERROR: Delivering is not a valid action in the " + this.getClass().getSimpleName() + " state.");
    }

    default void cancel(Order order) {
        System.out.println("ERROR: Cancelling is not a valid action in the " + this.getClass().getSimpleName() + " state.");
    }
}
